source/common/json/json_internal.cc

```diff
@@ -0,0 +1,668 @@
+      case Type::Integer:
+        value.push_back(element->value_.integer_value_);
+        break;
+      case Type::Null:
+        value.push_back(nlohmann::json::value_t::null);
+        break;
+      case Type::String:
+        value.push_back(element->value_.string_value_);
+      }
+    }
+    break;
+  }
+  case Type::Object: {
+    for (const auto& item : field.value_.object_value_) {
+      auto name = std::string(item.first);
+
+      switch (item.second->type_) {
+      case Type::Array:
+      case Type::Object: {
+        nlohmann::json nested_value;
+        buildJsonDocument(*item.second, nested_value);
+        value.emplace(name, nested_value);
+        break;
+      }
+      case Type::Boolean:
+        value.emplace(name, item.second->value_.boolean_value_);
+        break;
+      case Type::Double:
+        value.emplace(name, item.second->value_.double_value_);
+        break;
+      case Type::Integer:
+        value.emplace(name, item.second->value_.integer_value_);
+        break;
+      case Type::Null:
+        value.emplace(name, nlohmann::json::value_t::null);
+        break;
+      case Type::String:
+        value.emplace(name, item.second->value_.string_value_);
+        break;
+      }
+    }
+    break;
+  }
+  case Type::Null: {
+    break;
+  }
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+}
+
+nlohmann::json Field::asJsonDocument() const {
+  nlohmann::json j;
+  buildJsonDocument(*this, j);
+  return j;
+}
+
+uint64_t Field::hash() const { return HashUtil::xxHash64(asJsonString()); }
+
+bool Field::getBoolean(const std::string& name) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end() || !value_itr->second->isType(Type::Boolean)) {
+    throw Exception(fmt::format("key '{}' missing or not a boolean from lines {}-{}", name,
+                                line_number_start_, line_number_end_));
+  }
+  return value_itr->second->booleanValue();
+}
+
+bool Field::getBoolean(const std::string& name, bool default_value) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr != value_.object_value_.end()) {
+    return getBoolean(name);
+  } else {
+    return default_value;
+  }
+}
+
+double Field::getDouble(const std::string& name) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end() || !value_itr->second->isType(Type::Double)) {
+    throw Exception(fmt::format("key '{}' missing or not a double from lines {}-{}", name,
+                                line_number_start_, line_number_end_));
+  }
+  return value_itr->second->doubleValue();
+}
+
+double Field::getDouble(const std::string& name, double default_value) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr != value_.object_value_.end()) {
+    return getDouble(name);
+  } else {
```

Comment:
I think we prefer the form
```
if (exceptional-condition) {
  return exceptional-value;
}
return normal value;
```
But I can't find a style-guide ref for that. THere's a few different instances of that in this file. `Field::getDouble()` above is an example where you *are* doing it the way I think we prefer.

Reply:
existing style, fixed in the legacy code as well.